web/versioned/browsers-0.3.1.bzl

```diff
@@ -203,3 +203,10 @@ def org_mozilla_geckodriver():
             "GECKODRIVER": "geckodriver",
         },
     )
+
+def _browser_repositories_extension(ctx):
+    # TODO: This can be configured in MODULE.bazel via module extension tags.
```

Comment:
I was trying to figure this out. Does it need to be configured? Or could we enable all without consequences? (e.g. slower loading/builds, more downloads etc.)

Reply:
In case you missed this one.

Reply:
Yeah, I actually wants to implement this properly to show case how module extension works. Please see [this commit](https://github.com/bazelbuild/rules_webtesting/pull/478/commits/ecf708ee013a25ad7ffe8c896f4653cf8ad93576#diff-8f23351c622c1fa4118fe0289a67ee47c2445c8b7278f9e9ad2cf619573327dc).

It basically offers:
- a `version` tag to be used by any module to specify the version of browsers version. If multiple versions are specified, the highest one is chosen. (e.g module A requires 0.3.3, module B requires 0.3.4, then 0.3.4 is used eventually)
- a `override_version` tag to be used by the **root** module to pin the version. If a non-module root uses this tag, it'll be ignored.

Overall, this implements a reasonable resolution logic when there are version conflicts while also gives the root module full control when needed.

In addition, I removed all `firefox = False, chromium = False, sauce = False` parameters, because it's fine to declare all those repos, they will be fetched/downloaded lazily when the targets you build actually need them.

Let me know if this make sense for rules_webtesting.

Reply:
Looks like it matches the current functionality and provide in a nice way to handle conflict so +1.

I was wondering if it would be easy to allow the module to ask to alias a specific version so multiple could be used in parallel without conflict.

For example something like:
```
browser_repositories = use_extension("//web:extension.bzl", "browser_repositories_extension")
browser_repositories.install(version = "0.3.3", alias = "my_browsers")
use_repo(browser_repositories, "my_browsers")
```

Then user can do something like `@my_browsers//:chrome` without causing conflict.

Though it seems like that requires a much larger refactoring.

Reply:
> I was wondering if it would be easy to allow the module to ask to alias a specific version so multiple could be used in parallel without conflict.

So this module extension is mostly used to configure which version of browser repositories will be used by rules_webtesting. If other modules wants to use a specific version, they can still write a new module extension to call `browser_repositories` from a specific versioned bzl file like `web/versioned/browsers-0.3.1.bzl`, and it will be their "private" dependencies.
You can also call repository rule directly in MODULE.bazel via [`use_repo_rule`](https://bazel.build/rules/lib/repo/http#setup).